app/main/views/frameworks.py

```diff
@@ -529,6 +530,32 @@ def framework_supplier_declaration_overview(framework_slug):
 
         sections_errors[section.slug] = (section, errors)
 
+        # Create govukSummaryList-friendly fields
+        section.summary_list = []
+        for question in section.questions:
+            if sf["prefillDeclarationFromFrameworkSlug"] and section.prefill:
+                question.empty_message = "Review answer"
+            else:
+                question.empty_message = "Answer question"
+            section.summary_list.append(
+                to_summary_list_row(
+                    question,
+                    action_link=url_for(
+                        ".framework_supplier_declaration_edit",
+                        framework_slug=framework_slug,
+                        section_id=section.id,
+                        _anchor=question.id if section.questions[0].id != question.id else None
+                    ) if framework["status"] == "open" and sections_errors else None
+                )
+            )
+
+            # We need to parse key text for question references. This isn't optimal, but references only apply to
+            # this app, so holding off on centralising this logic to the Content Loader.
+            def label_question_reference(row):
+                row["key"]["text"] = question_references(row.get("key", {}).get("text", ""), section.get_question)
+                return row
+            section.summary_list = list(map(label_question_reference, section.summary_list))
```

Comment:
Currently this iterates through the list every time a question is added, did you mean to dedent this one level so it runs once after the for loop?

Reply:
Oof. Well, the good news is that it was loading speedily even though I hadn't caught this, so now it should be even quicker...

Thanks for spotting.